aws/structure_test.go

```diff
@@ -1322,3 +1322,188 @@ func TestNormalizeCloudFormationTemplate(t *testing.T) {
 		t.Fatalf("Got:\n\n%s\n\nExpected:\n\n%s\n", actual, validNormalizedYaml)
 	}
 }
+
+func TestCanonicalXML(t *testing.T) {
+	cases := []struct {
+		Name        string
+		Config      string
+		Expected    string
+		ExpectError bool
+	}{
+		{
+			Name:     "Config sample from MSDN",
+			Config:   testExampleXML_from_msdn,
+			Expected: testExampleXML_from_msdn,
+		},
+		{
+			Name:     "Config sample from MSDN, modified",
+			Config:   testExampleXML_from_msdn,
+			Expected: testExampleXML_from_msdn_modified,
+		},
+		{
+			Name:        "Config sample from MSDN, flaw",
+			Config:      testExampleXML_from_msdn,
+			Expected:    testExampleXML_from_msdn_flawed,
+			ExpectError: true,
+		},
+		{
+			Name: "A note",
+			Config: `
+<?xml version="1.0"?>
+<note>
+<to>You</to>
+<from>Me</from>
+<heading>Reminder</heading>
+<body>You're awesome</body>
+<rant/>
+<rant/>
+</note>
+`,
+			Expected: `
+<?xml version="1.0"?>
+<note>
+	<to>You</to>
+	<from>Me</from>
+	<heading>
+    Reminder
+    </heading>
+	<body>You're awesome</body>
+	<rant/>
+	<rant>
+</rant>
+</note>`,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.Name, func(t *testing.T) {
+			old, err := canonicalXML(tc.Config)
+			if err != nil {
+				t.Fatalf("Error getting canonical xml: %s", err)
```

Comment:
It may help if the error message mentioned this was error from _old_/_new_ XML.

Reply:
You ask too much


8e56a91c